topography_generator/utils/contours.hpp

```diff
@@ -90,7 +92,14 @@ void SimplifyContours(int simplificationZoom, Contours<ValueType> & contours)
       std::vector<m2::PointD> contourSimple;
       feature::SimplifyPoints(m2::SquaredDistanceFromSegmentToPoint(),
                               simplificationZoom, contour, contourSimple);
-      contour = std::move(contourSimple);
+      // Discard closed lines which are degenerate (<=3 points) or too small for the requested zoom level.
+      /// @todo it doesn't fix all cases as the simplification algo
+      /// can produce e.g. a 5 points closed but degenerate line ("flat", not a loop anymore).
```

Comment:
nit: Looks like this comment is outdated. We filter here all closed degenerated lines.